static/js/message_fetch.js

```diff
@@ -80,6 +80,11 @@ function get_messages_success(data, opts) {
     }
 
     if (opts.num_after > 0) {
+        const expected_latest_id = opts.msg_list.data.fetch_status.get_expected_max_message_id();
+        const present_latest_id = exports.get_frontfill_anchor(opts.msg_list);
```

Comment:
I'm not sure why we need to call `get_frontfill_anchor` here.  I had been imagining we'd do something with a cleaner interface, which is to pass `data.messages` into `finish_newer_batch`.  

Then the logic in `finish_newer_batch` can loop through the messages, find the max message ID in them, and compare that to `expected_latest_id`; and use that to decide how to set `fetch_status.has_found_newest`.

This would result in `get_expected_max_message_id` not needing to be part of the API/interface for the `fetch_status` module.